pre_commit/staged_files_only.py

```diff
@@ -47,7 +47,8 @@ def _unstaged_changes_cleared(patch_dir: str) -> Generator[None, None, None]:
         retcode=None,
     )
     if retcode and diff_stdout_binary.strip():
-        patch_filename = f'patch{int(time.time())}'
+        strftime = time.strftime('%Y-%m-%d-%H-%M-%S', time.gmtime())
+        patch_filename = f'patch-{git.get_basename()}-{strftime}'
```

Comment:
now they won't sort nicely inside the directory

Reply:
What if I remove the part with the basename and just keep the time ? The strftime is outputting 06 instead of 6 for month so I guess it would sort correctly without the basename.

Reply:
To be more precise this will sort correctly because the values are ordered from the largest to smallest unit of time: year, month, day, hour, minute, second. (This is iso 8601 without `:` and space so it works for a file sytem).